common/console.cpp

```diff
@@ -111,7 +112,16 @@ namespace console {
             }
         }
 
-        setlocale(LC_ALL, "");
+        auto locale = setlocale(LC_ALL, "");
+
+        if (locale == nullptr || !strcmp(locale, "C") || strcasestr(locale, "utf-8") == nullptr) {
+            auto lang = getenv("LANG");
+            if (lang != nullptr && strcmp(lang, "C") && strcasestr(lang, "utf-8") != nullptr) {
```

Comment:
You don't need to check both strcmp and strcasestr. On line 117 if the locale is "C", you don't need to check anything else. On line 119 if the locale contains "utf-8", you don't need to check anything else.

Reply:
This checks if locale (returned from `setlocale`) is either undefined, "C", or non-utf ( like "en" or "en_US" instead of "en_US.UTF-8"  )

And the second check is for `lang` (from env variable) to make sure it's a suitable candidate and doesn't have the same problem as locale

Afaik strcasestr (utf check) is necesary, because valid but non-utf locale still causes getwchar to fail on unicode input

Reply:
But C is also non-utf8, you don't need to special-case it.

Reply:
Oh!, Yes, you are absolutely right :)